src/detection_win.cpp

```diff
@@ -379,28 +382,28 @@ void BuildInitialDeviceList() {
 	}
 
 	SP_DEVINFO_DATA* pspDevInfoData = (SP_DEVINFO_DATA*) HeapAlloc(GetProcessHeap(), 0, sizeof(SP_DEVINFO_DATA));
-	pspDevInfoData->cbSize = sizeof(SP_DEVINFO_DATA);
-	for(int i=0; DllSetupDiEnumDeviceInfo(hDevInfo, i, pspDevInfoData); i++) {
-		DWORD nSize=0 ;
-		TCHAR buf[MAX_PATH];
+	if (pspDevInfoData) {
```

Comment:
Why can `pspDevInfoData` be undefined now?

Reply:
> If the function fails and you have not specified HEAP_GENERATE_EXCEPTIONS, the return value is NULL.
And below we check the pointer before release. I moved the check here.